compiler/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -216,6 +216,20 @@ impl<'a> FunctionContext<'a> {
                             self.codegen_array_checked(elements, typ[1].clone())?;
                         Tree::Branch(vec![slice_length.into(), slice_contents])
                     }
+                    ast::Type::Array(len, _) => {
+                        // A literal slice can be an array or a slice,
+                        // however we convert the array into a slice because printable types
+                        // do not handle slice array properly.
```

Comment:
Can you elaborate on this comment? A literal slice should only be a slice and never an array since we removed the polymorphism a while back

Reply:
Perhaps the issue is with `modulus_be_bytes` and possibly other builtin functions returning arrays when they should return slices or vice-versa